js/src/jupyter-leaflet.js

```diff
@@ -160,10 +202,27 @@ var LeafletMarkerView = LeafletUILayerView.extend({
         } else {
             this.obj.setOpacity(0);
         }
+        this.obj.setRotationAngle(this.model.get('rotation_angle'));
+        this.obj.setRotationOrigin(this.model.get('rotation_origin'));
+        this.listenTo(this.model, 'change:icon', function () {
+            this.set_icon(this.model.get('icon'));
+        }, this);        
+        this.set_icon(this.model.get('icon'));
     },
 });
 
 
+var LeafletIconView = LeafletUILayerView.extend({
+
+    create_obj: function () {
+        var that = this;
+
+        this.obj = L.icon(this.get_options());
+    },
+
```

Comment:
Does it work to change icon options dynamically? I mean, if you try something like:
```javascript
this.model.on('change:icon_url', () => {
    L.setOptions(this.obj, this.get_options());
});
```
Will the icon url be updated for each marker that uses this icon?

Reply:
I guess the answer is no... I took a look at the implementation of icons in LeafletJs, and I didn't see a way to update the icon options once it has been created. I think the only way to have a dynamic change is to remove the icon and create it again. We have the same problem with popups

Reply:
No the icon options can't be changed dynamically, but the marker can be assigned a different icon object  dynamically.   I don't see a setOptions method in leaflet.